add guide for creating a new pr (#191)
CodeYourFuture/curriculum: content/guides/create-a-pull-request/index.md

---
id: creating-a-pull-request
title: Creating a Pull Request
emoji: 📋
---

> # 🔃 Comparing changes

Reviewer: ```suggestion
## 🔃 Comparing changes
```

So it actually shows up in the table of contents - right now there's an empty TOC entry for bullet 1 (I'm not sure why)

As developers, we often develop new features on a branch.

> When we've completed the feature , we need to **merge** our changes into another branch.

Reviewer: ```suggestion
When we've completed the feature, we need to **merge** our changes into another branch.
```

So we'll need a way to compare changes between two different branches on Github.

Let's consider the following situation:

We have a repository like this:

### Original repository

```mermaid
---
title: CodeYourFuture/GitHomeworkTest
---
gitGraph
   commit
   commit
   commit
   commit
```

And there's another forked repository belonging to owner **Dedekind561**.
Dedekind561 has created a branch called `bio-page` with the new feature.

### 🍴 Forked repository

```mermaid
---
title: Dedekind561/GitHomeworkTest
---
gitGraph
   commit
   commit
   commit
   commit
   branch bio-page
   commit
   commit
   commit
```

> 🎯 Goal: Merge changes from the `bio-page` branch into the CodeYourFuture/GitHomeworkTest repo

## Pull Request

To merge changes from one branch to another, we need to let other users know👍

- what the changes are
- which branches we're comparing

> - additional context about the proposed change

Reviewer: ```suggestion
- additional context about the proposed change, like why we're making the change
```

For this purpose, we can use a **pull request**.

{{<note type="definition" title="definition: Pull Request">}}

A **pull request** is a request to merge code from one branch into another.

{{</note>}}

## Creating a new pull request

> During the course, we'll use Pull Requests to review your work. You'll need to submit your work on a branch and compare it

Reviewer: ```suggestion
During the course, we'll use Pull Requests to review your work. You'll need to push your work on a branch, and make it easy for volunteers to compare it.
```

We can use the Github interface to create a new Pull Request.

> The steps below assume you have a forked a repository called `GitHoneworkTest` and you've pushed a branch to this remote repository.

Reviewer: ```suggestion
The steps below assume you have a forked a repository called `GitHomeworkTest` and you've pushed a branch to this remote repository.
```

### Step 1

Go to your forked repository. The page should look like this:

![github-fork-main-page](github-fork-main-page.png)

> Double check your url, it should be `github.com/YOUR_USERNAME/GitHomeworkTest

Reviewer: ```suggestion
Double check your url, it should be `github.com/YOUR_USERNAME/GitHomeworkTest`
```

### Step 2

Locate the branches button on this page.

![github-fork-highlight-branches](github-fork-highlight-branches.png)

### Step 3

📝 Double check you're on a branches view like this.

![github-branches-view](github-branches-view.png)

### Step 4

Locate the **New pull request** button for the `bio-page` branch.

![highlight-new-pull-request](highlight-new-pull-request.png)

### Step 5

> Double check you cans see the **Open a pull request** view.

Reviewer: ```suggestion
Double check you can see the **Open a pull request** view.
```

![open-pull-request-page](open-pull-request-page.png)

#### Questions:

> a) Double check the base repository ( it should be set to `CodeYourFuture/GitHomeworkTest` )

Reviewer: ```suggestion
a) Double check the base repository - it should be set to the repository you're proposing merging the changes to (in this example, `CodeYourFuture/GitHomeworkTest`)
```

> b) Double check the head repository ( this should be the forked repository `YOUR_GITHUB_USERNAME/GitHomeworkTest` )

Reviewer: ```suggestion
b) Double check the head repository (this should be your forked repository `YOUR_GITHUB_USERNAME/GitHomeworkTest`)
```
